checks.d/kafka_consumer.py

```diff
@@ -47,20 +66,37 @@ def check(self, instance):
             # Query Zookeeper for consumer offsets
             consumer_offsets = {}
             topics = defaultdict(set)
+            if not consumer_groups:
+                consumer_groups = {consumer_group: None
+                                   for consumer_group in self._get_children(zk_conn,
+                                                                            zk_path_consumer,
+                                                                            'consumer groups')}
             for consumer_group, topic_partitions in consumer_groups.iteritems():
+                if topic_partitions is None:
+                    zk_path_topic = zk_path_topic_tmpl % (consumer_group)
+                    topic_partitions = self._get_children(zk_conn,
+                                                          zk_path_topic,
+                                                          'topics')
+                if isinstance(topic_partitions, list):
+                    topics = topic_partitions
+                    topic_partitions = {topic: self._get_partitions_for_topic(zk_conn,
+                                                                              zk_path_partition_tmpl,
+                                                                              consumer_group,
+                                                                              topic)
+                                        for topic in topics}
                 for topic, partitions in topic_partitions.iteritems():
                     # Remember the topic partitions that we've see so that we can
                     # look up their broker offsets later
                     topics[topic].update(set(partitions))
```

Comment:
This is now throwing:

    Traceback (most recent call last):
      File "/opt/datadog-agent/agent/checks/__init__.py", line 771, in run
        self.check(copy.deepcopy(instance))
      File "/opt/datadog-agent/agent/checks.d/kafka_consumer.py", line 90, in check
        topics[topic].update(set(partitions))
    TypeError: list indices must be integers, not str

I'm guessing when the topics are built automatically from ZK above they're built as a list when they should actually be a dict. Afraid I haven't time right now to further debug, but hopefully @dan-blanchard this can put you on the right track.

Looks like something above is passing